holoviews/core/spaces.py

```diff
@@ -672,7 +673,8 @@ def __getitem__(self, key):
 
         # Cache lookup
         try:
-            if util.dimensionless_contents(self.streams, self.kdims):
+            if (util.dimensionless_contents(self.streams, self.kdims) and
+                not self._stream_cache_lookup):
                 raise KeyError('Using dimensionless streams disables DynamicMap cache')
```

Comment:
From our discussion, we think it ought to be something like:

```
if (util.dimensionless_contents(self.streams, self.kdims)
                  raise KeyError('Using dimensionless streams disables DynamicMap cache')
elif self._disable_cache_lookup:
                  raise KeyError('Cache lookup disabled')
```

That way the variable isn't about streams and you can disable cache lookup on dynamic maps not matter how they are (or aren't) using streams.